app/assets/javascripts/app/calendar.js

```diff
@@ -133,12 +138,87 @@ window.FullCalendarConfig = class FullCalendarConfig {
   }
 
   formattedEventPeriod(event) {
+    let format = (this.isDailyBooking() && event.end.isAfter(event.start, 'day')) ? "MM/DD/YY h:mmA" : "h:mmA";
+
     return [event.start, event.end].
-      map(date => $.fullCalendar.formatDate(date, "h:mmA")).
-      join("&ndash;");
+      map(date => $.fullCalendar.formatDate(date, format)).
+      join(" &ndash; ");
   }
 
   linkToEditOrder(event) {
     if ((event.orderId != null) && (typeof orders_path_base !== 'undefined' && orders_path_base !== null)) { return `<a href='${orders_path_base}/${event.orderId}'>Edit</a>`; }
   }
+
+  /*
+   * Render monthly view events with margins
+   * depending on the start and end offset
+   * from midnight.
+   */
+  adjustEvent(event, element, view) {
+    // Do nothing if not daily booking
+    if (!this.isDailyBooking()) { return; }
+    // Don't apply changes unless monthly view
+    if (view.name != 'month') { return; }
+    // exclude allDay and background events
+    if (event['rendering'] == 'background' || event['allDay']) { return; }
+
+    let seg = $(element).data('fc-seg');
+    // if there's no info about the drawn event segment
+    // there's nothing to do
+    if (!seg) { return; }
+
+    let startOfDay = event.start.clone();
+    startOfDay.startOf('day');
+
+    let endOfDay = event.end.clone();
+    endOfDay.startOf('day');
+    // If event end is the start of the day it's already
+    // the follwoing day
+    if (endOfDay.diff(event.end) != 0) {
+      endOfDay.add(1, 'day')
+    }
+
+    let startOffset = event.start.diff(startOfDay, 'minutes');
+    if (!seg.isStart) {
+      // Event started some row above
```

Comment:
The calendar is rendered as a table where each row is a week. Each segment represents the part of an event that occurs in a specific week.

The attributes `isStart` and `isEnd` tell you if the event starts/ends on the week the segment is representing, for example if theres and event lasting 8 days then we would have at least two segments for that event because it starts on one week and ends the following. This function is called once for each segment.

Reply:
Can we maybe have a small comment for that?

Reply:
I'll leave a link to the documentation